tests/NuGet.Services.AzureSearch.FunctionalTests/BasicTests/V2SearchProtocolTests.cs

```diff
@@ -0,0 +1,266 @@
+            var resultsWithoutSkip = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Take = 10 });
+            var resultsWithSkip = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Skip = skip, Take = 10 - skip });
+
+            Assert.NotNull(resultsWithoutSkip);
+            Assert.NotNull(resultsWithSkip);
+            Assert.True(resultsWithoutSkip.Data.Count > 1);
+            Assert.True(resultsWithSkip.Data.Count > 1);
+            Assert.True(resultsWithoutSkip.Data.Count == resultsWithSkip.Data.Count + skip);
+            var packageIDListThatShouldBeSkipped = resultsWithoutSkip.Data
+                .Select(x => x.PackageRegistration.Id)
+                .Take(skip);
+            var skippedResultPackageIdList = resultsWithSkip.Data
+                .Select(x => x.PackageRegistration.Id);
+            var commonResults = skippedResultPackageIdList
+                .Where(x => packageIDListThatShouldBeSkipped.Contains(x, StringComparer.OrdinalIgnoreCase));
+            Assert.True(commonResults.Count() == 0);
+        }
+
+        [Fact]
+        public async Task CountOnlyReturnsCountOfSearchResults()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { CountOnly = true });
+
+            Assert.NotNull(results);
+            Assert.True(results.TotalHits > 1);
+            Assert.Null(results.Data);
+        }
+
+        [Fact]
+        public async Task ResultsHonorPreReleaseField()
+        {
+            var searchTerm = "json";
+
+            var resultsWithPrerelease = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Prerelease = true }); ;
+            Assert.NotNull(resultsWithPrerelease);
+            Assert.True(resultsWithPrerelease.Data.Count > 1);
+
+            var resultsWithoutPrerelease = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Prerelease = false });
+            Assert.NotNull(resultsWithoutPrerelease);
+            Assert.True(resultsWithoutPrerelease.Data.Count > 1);
+
+            // The result count should be different for included prerelease results, 
+            // else it means the search term responded with same results i.e. no results have prerelease versions
+            Assert.True(resultsWithPrerelease.TotalHits > resultsWithoutPrerelease.TotalHits, 
+                $"The search term {searchTerm} does not seem to have any prerelease versions in the search index.");
+
+            var hasPrereleaseVersions = resultsWithPrerelease
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.True(hasPrereleaseVersions);
+
+            hasPrereleaseVersions = resultsWithoutPrerelease
+                .Data
+                .Any(x => IsPrerelease(x.Version));
+            Assert.False(hasPrereleaseVersions);
+        }
+
+        [Fact]
+        public async Task ResultsHonorSemverLevel()
+        {
+            var searchTerm = "packageId:BaseTestPackage.SemVer2";
```

Comment:
I like that this is simple and low overhead, but this package may be deleted/unlisted without us realizing this will break this functional test. To mitigate this concern, what do you think of renaming this package to something like `TestPackageDoNotRemove.SemVer2`, and then making that package's description a link to this test?

Reply:
Also, please follow the `Contants.TestPackageId` pattern

Reply:
We do use `BaseTestPackage` for our tests, hence the similar name. I would leave it as is. If it does end up breaking in future I would create one, but I think we've been fine for now.